### ISSUE TITLE
Storybook does not accept headless-ui component types

### ISSUE THREAD

**Issue by marcobusemann** [tailwindlabs/headlessui#1659]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

For example: v1.6.6

**Describe your issue**

We are using Storybook to showcase different ui components. Since we upgraded from v1.4.2 to 1.6.6, the headless-ui component types are no longer compatible with Storybook. 

```
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import { Popover } from '@headlessui/react'

export default {
  title: 'Common/Popover',
  component: Popover,
} as ComponentMeta<typeof Popover>
``` 

This now shows an error for `ComponentMeta<typeof Popover>`:

> Type '(<TTag extends ElementType<any> = "div">(props: Props<TTag, PopoverRenderPropArg, "1D45E01E-AF44-47C4-988A-19A94EBAF55C">, ref: Ref<HTMLElement>) => Element) & { ...; } & { ...; }' does not satisfy the constraint 'keyof IntrinsicElements | JSXElementConstructor<any>'.
  Type '(<TTag extends ElementType<any> = "div">(props: Props<TTag, PopoverRenderPropArg, "1D45E01E-AF44-47C4-988A-19A94EBAF55C">, ref: Ref<HTMLElement>) => Element) & { ...; } & { ...; }' is not assignable to type '(props: any) => ReactElement<any, any> | null'.ts(2344)

So basically `IntrinsicElements` or `JSXElementConstructor<any>`  is missing on the type-declareation.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

I'm going to close this as I believe this is a duplicate of #1365
